nxdrive/dao/base.py

```diff
@@ -227,16 +236,20 @@ def dispose(self) -> None:
     def _get_write_connection(self) -> Connection:
         if self.in_tx:
             if self.conn is None:
-                self._create_main_conn()
+                self.conn = self._create_main_conn()
             return self.conn
         return self._get_read_connection()
 
     def _get_read_connection(self) -> Connection:
         # If in transaction
         if self.in_tx is not None:
             if current_thread_id() == self.in_tx:
-                # Return the write connection
-                return self.conn
+                if self.conn:
+                    # Return the write connection
+                    return self.conn
+                elif self.conn is None:
```

Comment:
We can simplify the check here, right?
```suggestion
                else:
```